From 8fc6b0906839d2cb358e11bd481b770a82e16f6b Mon Sep 17 00:00:00 2001
From: Abhishek <apatgaonkar@habana.ai>
Date: Wed, 30 Jun 2021 05:39:19 +0300
Subject: [PATCH] [SW-25060] greco: add sp polling registers

To avoid interrupt storm from LKD to host, GIC irq controller
and ARC SW irq will only be available for privileged entity.
LKD has to use scratchpad registers to generate host irqs.
To provide these register offsets to LKD, added them to dynamic
cpu register struct.

Change-Id: I347ad8f135921e3e7490db486431ad3dc1f7f71d
---

diff --git a/arch/greco/asic.c b/arch/greco/asic.c
index a94a3f1..3576b67 100644
--- a/arch/greco/asic.c
+++ b/arch/greco/asic.c
@@ -134,4 +134,13 @@
 	regs->hw_state = mmHW_STATE;
 	regs->kmd_msg_to_cpu = mmPSOC_GLOBAL_CONF_KMD_MSG_TO_CPU;
 	regs->cpu_cmd_status_to_host = mmCPU_CMD_STATUS_TO_HOST;
+	regs->gic_tpc_qm_irq_ctrl = mmGIC_TPC_QM_IRQ_CTRL_POLL_REG;
+	regs->gic_mme_qm_irq_ctrl = mmGIC_MME_QM_IRQ_CTRL_POLL_REG;
+	regs->gic_dma_qm_irq_ctrl = mmGIC_DMA_QM_IRQ_CTRL_POLL_REG;
+	regs->gic_rot_qm_irq_ctrl = mmGIC_ROT_QM_IRQ_CTRL_POLL_REG;
+	regs->gic_dma_core_irq_ctrl = mmGIC_DMA_CR_IRQ_CTRL_POLL_REG;
+	regs->gic_host_pi_upd_irq = mmGIC_HOST_PI_UPD_IRQ_POLL_REG;
+	regs->gic_host_halt_irq = mmGIC_HOST_HALT_IRQ_POLL_REG;
+	regs->gic_host_ints_irq  = mmGIC_HOST_INTS_IRQ_POLL_REG;
+	regs->gic_host_soft_rst_irq = mmGIC_HOST_SOFT_RST_IRQ_POLL_REG;
 }
